### ISSUE TITLE
rc41 fails to launch JNLP agents

### ISSUE THREAD

**Issue by samrocketman** [KostyaSha/yet-another-docker-plugin#197]
# Problem

Provisioning JNLP agents no longer works after upgrading Jenkins and Yet-another-docker-plugin.

# Jenkins master and plugin versions

<details><summary>Jenkins master and plugin versions (click to expand)</summary>

```
Jenkins Core 2.73.1
ace-editor 1.1
ant 1.7
antisamy-markup-formatter 1.5
authentication-tokens 1.3
blueocean 1.2.4
blueocean-autofavorite 1.0.0
blueocean-bitbucket-pipeline 1.2.4
blueocean-commons 1.2.4
blueocean-config 1.2.4
blueocean-dashboard 1.2.4
blueocean-display-url 2.1.0
blueocean-events 1.2.4
blueocean-git-pipeline 1.2.4
blueocean-github-pipeline 1.2.4
blueocean-i18n 1.2.4
blueocean-jira 1.2.4
blueocean-jwt 1.2.4
blueocean-personalization 1.2.4
blueocean-pipeline-api-impl 1.2.4
blueocean-pipeline-editor 1.2.4
blueocean-pipeline-scm-api 1.2.4
blueocean-rest 1.2.4
blueocean-rest-impl 1.2.4
blueocean-web 1.2.4
bouncycastle-api 2.16.2
branch-api 2.0.11
cloud-stats 0.13
cloudbees-bitbucket-branch-source 2.2.3
cloudbees-folder 6.1.2
cobertura 1.11
console-column-plugin 1.5
covcomplplot 1.1.1
credentials 2.1.16
credentials-binding 1.13
cvs 2.13
dashboard-view 2.9.11
display-url-api 2.0
docker-commons 1.8
docker-workflow 1.13
durable-task 1.14
embeddable-build-status 1.9
external-monitor-job 1.7
favorite 2.3.0
git 3.5.1
git-client 2.5.0
git-server 1.7
github 1.28.0
github-api 1.86
github-branch-source 2.2.3
github-oauth 0.27
groovy 2.0
groovy-label-assignment 1.2.0
handlebars 1.1.1
htmlpublisher 1.14
icon-shim 2.0.3
jackson2-api 2.7.3
javadoc 1.4
jira 2.4.2
job-dsl 1.65
jquery-detached 1.2.1
junit 1.21
ldap 1.17
mailer 1.20
mapdb-api 1.0.9.0
mask-passwords 2.10.1
matrix-auth 1.7
matrix-project 1.11
maven-plugin 2.17
mercurial 2.1
momentjs 1.1.1
pam-auth 1.3
pipeline-build-step 2.5.1
pipeline-graph-analysis 1.5
pipeline-input-step 2.8
pipeline-milestone-step 1.3.1
pipeline-model-api 1.2
pipeline-model-declarative-agent 1.1.1
pipeline-model-definition 1.2
pipeline-model-extensions 1.2
pipeline-rest-api 2.9
pipeline-stage-step 2.2
pipeline-stage-tags-metadata 1.2
pipeline-stage-view 2.9
plain-credentials 1.4
pubsub-light 1.12
rich-text-publisher-plugin 1.4
scm-api 2.2.2
script-security 1.34
sse-gateway 1.15
ssh-credentials 1.13
ssh-slaves 1.21
structs 1.10
subversion 2.9
token-macro 2.3
translation 1.15
variant 1.1
view-job-filters 1.27
windows-slaves 1.3.1
workflow-aggregator 2.5
workflow-api 2.20
workflow-basic-steps 2.6
workflow-cps 2.40
workflow-cps-global-lib 2.9
workflow-durable-task-step 2.15
workflow-job 2.14.1
workflow-multibranch 2.16
workflow-scm-step 2.6
workflow-step-api 2.13
workflow-support 2.14
yet-another-docker-plugin 0.1.0-rc41
```

</details>

# Jenkins master log

<details><summary>Master log of provisioning the container (click to expand)</summary>

```
Sep 23, 2017 12:00:26 PM INFO hudson.WebAppMain$3 run

Jenkins is fully up and running

Sep 23, 2017 12:02:04 PM INFO com.github.kostyasha.yad.DockerCloud provision

Asked to provision load: '1', for: 'stable&&docker&&ubuntu1604&&sudo&&"language:groovy"&&env&&jdk' label

Sep 23, 2017 12:02:04 PM INFO com.github.kostyasha.yad.DockerCloud provision

Will provision 'jervis-docker-jvm:latest', for label: 'stable&&docker&&ubuntu1604&&sudo&&"language:groovy"&&env&&jdk', in cloud: 'docker-local'

Sep 23, 2017 12:02:06 PM INFO com.github.kostyasha.yad.DockerCloud addProvisionedSlave

Provisioning 'jervis-docker-jvm:latest' number '0' on 'docker-local'; Total containers: '0'

Sep 23, 2017 12:02:14 PM INFO hudson.slaves.NodeProvisioner$2 run

jervis-docker-jvm:latest provisioning successfully completed. We have now 2 computer(s)

Sep 23, 2017 12:02:14 PM INFO com.github.kostyasha.yad.launcher.DockerComputerJNLPLauncher launch

Starting connection command for f7ce780f9fe71d0fd16f5393b8db1769f7efc0735076bf66f152e83406d87377
```

</details>

# Failed Docker container logs

<details><summary>Container logs of failed starting container (click this to expand)</summary>

Here's the docker container log of the container `f7ce780f9fe71d0fd16f5393b8db1769f7efc0735076bf66f152e83406d87377` failing to start.  Note: I don't care about secrets because this local environment is dev on my laptop and gets regularly deleted.

```
+ cat
+ chmod +x /tmp/init.sh
+ exec /tmp/init.sh
+ export CONFIG=/tmp/config.sh
+ [ ! -f /tmp/config.sh ]
+ echo No config, sleeping for 1 second
No config, sleeping for 1 second
+ sleep 1
+ [ ! -f /tmp/config.sh ]
+ echo No config, sleeping for 1 second
No config, sleeping for 1 second
+ sleep 1
+ [ ! -f /tmp/config.sh ]
+ echo No config, sleeping for 1 second
No config, sleeping for 1 second
+ sleep 1
+ [ ! -f /tmp/config.sh ]
+ echo No config, sleeping for 1 second
No config, sleeping for 1 second
+ sleep 1
+ [ ! -f /tmp/config.sh ]
+ echo No config, sleeping for 1 second
No config, sleeping for 1 second
+ sleep 1
+ [ ! -f /tmp/config.sh ]
+ echo No config, sleeping for 1 second
No config, sleeping for 1 second
+ sleep 1
+ [ ! -f /tmp/config.sh ]
+ echo No config, sleeping for 1 second
No config, sleeping for 1 second
+ sleep 1
+ [ ! -f /tmp/config.sh ]
+ echo No config, sleeping for 1 second
No config, sleeping for 1 second
+ sleep 1
+ [ ! -f /tmp/config.sh ]
+ echo Found config file
Found config file
+ . /tmp/config.sh
+ JENKINS_URL=http://172.17.0.1:8080
+ JENKINS_USER=jenkins
+ JENKINS_HOME=/home/jenkins
+ COMPUTER_URL=computer/docker-local-f7ce780f9fe7/
+ COMPUTER_SECRET=51cd0614257af1d89e9564269afb9f3d214c3d7d52ec4d16f64227b195958727
+ JAVA_OPTS=
+ SLAVE_OPTS=
+ NO_CERTIFICATE_CHECK=false
+ NO_RECONNECT_SLAVE=true
+ [ -z http://172.17.0.1:8080 ]
+ [ -z computer/docker-local-f7ce780f9fe7/ ]
+ [ -z /home/jenkins ]
+ id -u jenkins
1000
+ [ ! -d /home/jenkins ]
+ stat /home/jenkins
  File: '/home/jenkins'
  Size: 175       	Blocks: 0          IO Block: 4096   directory
Device: 25h/37d	Inode: 77758       Links: 6
Access: (0755/drwxr-xr-x)  Uid: ( 1000/ jenkins)   Gid: ( 1000/ jenkins)
Access: 2017-09-21 01:01:03.074436917 +0000
Modify: 2017-09-21 01:01:02.104435645 +0000
Change: 2017-09-21 01:01:02.104435645 +0000
 Birth: -
+ cd /home/jenkins
+ [ false = true ]
+ WGET_OPTIONS=
+ CURL_OPTIONS=
+ NO_SLAVE_CERT=
+ command -v wget
+ [ -x  ]
+ command -v curl
+ [ -x /usr/bin/curl ]
+ curl --remote-name http://172.17.0.1:8080/jnlpJars/slave.jar
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  713k  100  713k    0     0  39.1M      0 --:--:-- --:--:-- --:--:-- 43.5M
+ env
HOSTNAME=f7ce780f9fe7
HOME=/home/jenkins
OLDPWD=/home/jenkins
TERM=xterm
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
CONFIG=/tmp/config.sh
PWD=/home/jenkins
+ RUN_CMD=java
+ [ -n  ]
+ RUN_CMD=java -jar slave.jar
+ [ true == true ]
/tmp/init.sh: 89: [: true: unexpected operator
+ RUN_CMD=java -jar slave.jar
+ [ -n  ]
+ RUN_CMD=java -jar slave.jar -jnlpUrl http://172.17.0.1:8080/computer/docker-local-f7ce780f9fe7//slave-agent.jnlp
+ [ -n 51cd0614257af1d89e9564269afb9f3d214c3d7d52ec4d16f64227b195958727 ]
+ RUN_CMD=java -jar slave.jar -jnlpUrl http://172.17.0.1:8080/computer/docker-local-f7ce780f9fe7//slave-agent.jnlp -secret 51cd0614257af1d89e9564269afb9f3d214c3d7d52ec4d16f64227b195958727
+ id -nu
+ [ jenkins != jenkins ]
+ eval java -jar slave.jar -jnlpUrl http://172.17.0.1:8080/computer/docker-local-f7ce780f9fe7//slave-agent.jnlp -secret 51cd0614257af1d89e9564269afb9f3d214c3d7d52ec4d16f64227b195958727
+ java -jar slave.jar -jnlpUrl http://172.17.0.1:8080/computer/docker-local-f7ce780f9fe7//slave-agent.jnlp -secret 51cd0614257af1d89e9564269afb9f3d214c3d7d52ec4d16f64227b195958727
Exception in thread "main" java.io.IOException: The specified remoting internal directory should be fully accessible to the remoting executable (RWX): /
	at org.jenkinsci.remoting.engine.WorkDirManager.verifyDirectory(WorkDirManager.java:232)
	at org.jenkinsci.remoting.engine.WorkDirManager.initializeWorkDir(WorkDirManager.java:190)
	at hudson.remoting.Engine.startEngine(Engine.java:252)
	at hudson.remoting.Engine.startEngine(Engine.java:229)
	at hudson.remoting.jnlp.Main.main(Main.java:217)
	at hudson.remoting.jnlp.Main._main(Main.java:212)
	at hudson.remoting.Launcher.run(Launcher.java:351)
	at hudson.remoting.Launcher.main(Launcher.java:277)
```

</details>

**Comment by samrocketman** [comment]
This duplicates #184.  Looks like this issue now affects LTS which is supposed to be stable Jenkins.

**Comment by KostyaSha** [comment]
so, 
+ stat /home/jenkins
  File: '/home/jenkins'
  Size: 175       	Blocks: 0          IO Block: 4096   directory
Device: 25h/37d	Inode: 77758       Links: 6
Access: (0755/drwxr-xr-x)  Uid: ( 1000/ jenkins)   Gid: ( 1000/ jenkins)

has correct perms and slaves work fine before, but now they fails. `/` should not be RWx accessible, this error smells like remoting bug @oleg-nenashev

**Comment by VQuery** [comment]
@KostyaSha  facing same issue in docker plugin .Could you please do the needful?

**Comment by webczat** [comment]
same here

**Comment by samrocketman** [comment]
@VQuery and @webczat as noted in https://github.com/KostyaSha/yet-another-docker-plugin/issues/184#issuecomment-322996960

> As a workaround, you can define the workdir at the JNLP settings, Slave (slave.jar) options:
> ```-workDir /home/jenkins```

**Comment by webczat** [comment]
isn't the issue fixed in newest version of jenkins (still unreleased probably)?

**Comment by oleg-nenashev** [comment]
@webczat The fix has been backported to 2.73.3. The release candidate testing is underway, you can pick the RC build here: https://groups.google.com/forum/#!topic/jenkinsci-dev/MOzX503CVdU. ETA - ~Nov 9

**Comment by webczat** [comment]
well actually I believe that the jenkins release cycle is fast enough so that I can just wait a little to receive it, as I do not urgently need to execute docker builds. Although I cannot properly test my docker template before this release, but I should remember.
